summarize_pr.py

```diff
@@ -42,29 +40,113 @@ def summarize_pr(
                     f"Raw File Content:\n```\n{response.text}\n```\n\n"
                 )
 
+    return pull_request_content
+
+
+def _process_code_base(
+    pull_request_content: str, model_id: str, task_prompt: str, post_prompt: str = ""
+):
     model = GenerativeModel(
         model_id,
         system_instruction=[
             "You are an expert software engineer.",
         ],
+        generation_config=GenerationConfig(
+            max_output_tokens=8192,
+        ),
     )
-
     prompt = [
-        "The following is the content of a GitHub Pull Request for a repository focused on Generative AI with Google Cloud. This content includes the Pull Request title, Pull Request description, a list of all of the files changed with the file name, the code diff and the raw file content. Your task is to output a summary of the Pull Request in Markdown format.",
-        "Content:",
+        "The following is the content of a GitHub Pull Request for a repository focused on Generative AI with Google Cloud. This content includes the Pull Request title, Pull Request description, a list of all of the files changed with the file name, the code diff and the raw file content.",
+        task_prompt,
+        "Pull Request Content:",
         pull_request_content,
-        "Summary:",
+        post_prompt,
     ]
 
     print("---Prompt---\n", prompt)
     response = model.generate_content(prompt)
     print("---Gemini Response---\n", response)
+    return response.text
+
+
+def analyze_pr(api: GhApi, pr_number: str, model_id: str) -> Tuple:
+    pull_request_content = _get_pr_content(api, pr_number)
+
+    summarize_output = _process_code_base(
+        pull_request_content,
+        model_id,
+        "Your task is to summarize the Pull Request in Markdown format.",
+        post_prompt="Summary:",
+    )
+
+    analysis_output = _process_code_base(
+        pull_request_content,
+        model_id,
+        task_prompt="Your task is to analyze each code file line by line, then output any bugs you find and improvements that can be made to the code quality for readability and maintainability.",
+        post_prompt="Analysis:",
+    )
+
+    latest_commit = api.pulls.list_commits(pr_number)[-1]["sha"]
+
+    # Create a comment on the PR
+    comment_content = f"""# Pull Request Summary from Gemini ✨\n{summarize_output}\n\n# Code Analysis from Gemini ✨\n{analysis_output}\n Generated at Commit: `{latest_commit}`"""
 
     # Post the summary as a comment
-    api.issues.create_comment(
-        pr_number, f"## Pull Request Summary from Gemini\n {response.text}"
+    api.issues.create_comment(pr_number, comment_content)
+
+    return summarize_output, analysis_output
+
+
+def convert_to_sarif(model_id: str, analysis_content: str) -> str:
+    model = GenerativeModel(
+        model_id,
+        system_instruction=[
+            "You are an expert software engineer.",
+        ],
     )
 
```

Comment:
I recommend adding a `try...except` block to handle potential errors during the API call. This will make the script more robust and prevent unexpected failures.